src/views/view_sycl.hpp

```diff
@@ -143,22 +143,39 @@ struct VectorView<
    */
   SYCL_BLAS_INLINE increment_t get_stride() const { return strd_; }
 
-  SYCL_BLAS_INLINE scalar_t eval(index_t i) const {
+  /**** EVALUATING ****/
+  template <bool use_as_ptr = false>
+  SYCL_BLAS_INLINE typename std::enable_if<!use_as_ptr, scalar_t &>::type eval(
+      index_t i) {
     return (strd_ == 1) ? *(ptr_ + i) : *(ptr_ + i * strd_);
   }
-  /**** EVALUATING ****/
-  SYCL_BLAS_INLINE scalar_t &eval(index_t i) {
+
+  template <bool use_as_ptr = false>
+  SYCL_BLAS_INLINE typename std::enable_if<!use_as_ptr, const scalar_t>::type
```

Comment:
Does this return type need to be const? It's a copy either way so the const seems superfluous here unless required.

Reply:
It probably doesn't, I did that for consistency with the existing code. @mehdi-goli any opinion on this?

Reply:
As it is a copy const qualifier has no effect here, however, the function labeled as const, if it does not give you an error when removing the const, then remove it

Reply:
Fixed, I've removed the useless `const` qualifiers in the return type. The function **has to** be labeled as `const` though because 1) otherwise it doesn't compile as there would be two overloads that differ only in their return type ; 2) it is good to label a method as `const` if it doesn't modify the object state.